internal/providers/msentraid/msentraid.go

```diff
@@ -108,6 +108,10 @@ func (p Provider) getGroups(token *oauth2.Token) ([]info.Group, error) {
 	if err != nil {
 		return nil, err
 	}
+	if m == nil {
+		slog.Debug("No groups found for user")
```

Comment:
Maybe use something like "Got empty object when fetching user memberships, assuming that user has no groups"?

The more descriptive the debug message, the better for our future selves when trying to understand what's breaking.

Reply:
In my opinion the main purpose of such a debug line (which doesn't print any values) is to find the place in the code where it's printed, which allows us to better understand what is breaking than we could possibly explain in the message itself. Nevertheless, I tried to make it more descriptive.

Reply:
For us, yes. But we do guide users on how to enable the debug logs and this could help them figure out the issue by themselves as, if we get into this case, it's likely a problem in the provider configuration.